src/sentry/scim/endpoints/teams.py

```diff
@@ -318,15 +318,17 @@ def _rename_team_operation(self, request: Request, new_name, team):
             data={"name": new_name, "slug": slugify(new_name)},
             partial=True,
         )
-        if serializer.is_valid():
-            team = serializer.save()
-            self.create_audit_entry(
-                request=request,
-                organization=team.organization,
-                target_object=team.id,
-                event=audit_log.get_event_id("TEAM_EDIT"),
-                data=team.get_audit_log_data(),
-            )
+        if not serializer.is_valid():
+            raise serializers.ValidationError(serializer.errors)
```

Comment:
Before we didn't raise error on invalid serializer? 🤔

Reply:
Let's make sure there is not a good reason for silently failing. I'm not familiar with SCIM. @cathteng do you know?

Reply:
we check the validity of the `SCIMTeamPatchRequestSerializer` serializer in `patch()` where this function is being called. i think it's good to check here too for `TeamSerializer` because of the new constraint on slug names